src/FffGcodeWriter.cpp

```diff
@@ -1602,8 +1602,113 @@ void FffGcodeWriter::processSpiralizedWall(const SliceDataStorage& storage, Laye
     const bool is_top_layer = ((size_t)layer_nr == (storage.spiralize_wall_outlines.size() - 1) || storage.spiralize_wall_outlines[layer_nr + 1] == nullptr);
     ConstPolygonRef wall_outline = part.insets[0][0]; // current layer outer wall outline
     const int seam_vertex_idx = storage.spiralize_seam_vertex_indices[layer_nr]; // use pre-computed seam vertex index for current layer
+
+    std::vector<float> flows(wall_outline.size(), 1.0); // flow multiplier for each line segment
+    std::vector<Point> shifts(wall_outline.size()); // per vertex shift to compensate for line width changes
+
+    const coord_t default_line_width = mesh.settings.get<coord_t>("wall_line_width_0");
+    const double min_line_width = mesh.settings.get<Ratio>("spiralize_min_line_width");
+    const double max_line_width = mesh.settings.get<Ratio>("spiralize_max_line_width");
+
+    if (min_line_width != 1.0 || max_line_width != 1.0)
+    {
+        // wall line width can vary
+
+        auto get_line_width = [&](const Point& prev_point, const Point& this_point, const Point& next_point, coord_t& line_width, Point& bisector, double& abs_sine)
+        {
+            const double corner_rads = LinearAlg2D::getAngleLeft(prev_point, this_point, next_point);
+
+            bisector = rotate(normal(next_point - this_point, default_line_width * 5), corner_rads / 2);
+            abs_sine = std::abs(std::sin(corner_rads / 2));
+
+            Polygons lines;
+            lines.addLine(this_point - bisector, this_point + bisector);
+#if 0
+            // diagnostic - print vertex bisector lines
+            gcode_layer.addTravel(lines[0][0]);
+            gcode_layer.addExtrusionMove(lines[0][1], this_point == wall_outline[seam_vertex_idx] ? mesh_config.insetX_config : mesh_config.inset0_config, SpaceFillType::Lines, 0.1);
+#endif
+
+            lines = part.outline.intersectionPolyLines(lines);
+            if (lines.size() > 0)
+            {
+                // find clipped line segment that starts/ends close to poly[n]
+                unsigned ln = 0;
+                while (ln < (lines.size() - 1) && vSize2(lines[ln][0] - this_point) > 100 && vSize2(lines[ln][1] - this_point) > 100)
+                {
+                    ++ln;
+                }
+
+                line_width = vSize(lines[ln][1] - lines[ln][0]) * abs_sine;
+            }
+        };
+
+        for (unsigned n = 0; n < wall_outline.size(); ++n)
+        {
+            const Point& this_point = wall_outline[n];
+            const Point& prev_point = wall_outline[(n + wall_outline.size() - 1) % wall_outline.size()];
+            const Point& next_point = wall_outline[(n + 1) % wall_outline.size()];
+
+            Point bisector;
+            double abs_sine = 1;
+            coord_t line_width = default_line_width;
+
+            get_line_width(prev_point, this_point, next_point, line_width, bisector, abs_sine);
```

Comment:
I think it's a bit cleaner to use a return statement (instead of an output parameter, since they are harder to understand what is being modified and what isn't.

Reply:
Its actually producing 3 values: line_width, bisector and abs_sine.

Reply:
In that case, it's not just getting the line_width, so I guess we need to find a better name. A function like "get_line_width" to me feels like it returns a width (since thats what it's naming implies)